Protocol.md

```diff
@@ -0,0 +1,187 @@
+# DLC Protocol
+
+This document describes a protocol for two parties to set up a Discreet Log Contract (DLC).
+
+## Preliminaries
+
+The following protocol assumes that both parties have already agreed on the contract terms (the possible outputs of the contract).
+Negotiation of the contract, or decisions on the underlying asset are thus not part of the exchanged information.
+
+## Description
+
+The two parties exchange a set of three messages, and terminates when the non-initiating party broadcasts the Fund Transaction to the Bitcoin network.
+The following figure gives a brief overview of the protocol:
+
+    +-------+                    +-------+
+    |       |                    |       |
+    |       |---- (1) offer  --->|       |
+    |       |                    |       |
+    |       |<--- (2) accept ----|       |
+    |   A   |                    |   B   |
+    |       |---- (3) sign   --->|       |
+    |       |                    |       |
+    |       |                    |  (4) broadcast fund-tx
+    |       |                    |       |
+    +-------+                    +-------+
+
+### Offer
+
+The initiating party starts the protocol by sending an `offer` message to the other party.
+This message includes the following information:
+1. Contract information (an array of triples defined below)
+1. Oracle information (optional)
+1. A's fund amount (satoshi)
+1. A's [extended public key](https://github.com/discreetlogcontracts/dlcspecs/blob/master/KeyDerivation.md)
+1. A's inputs
+1. A's change address
+1. Estimatesmartfee (satoshi/vbyte)
+1. CET CSV delay
+1. CET locktime (corresponding to the contract maturity date)
```

Comment:
Yeah I guess we need to decide what we prefer here. I would personally prefer to have all the necessary information required to build the transactions be transmitted, because if we rely on implementation specific communication I feel we kind of miss the point of having compatibility across implementations.

Reply:
Yeah, I think I'm open to changing this in the future but that it should be here for now, just thought I'd leave a note :+1: